### PR TITLE
fix client hidden service connections

### PR DESCRIPTION
**Pull request by arades79**

### CODE REVIEW

File: hyperdome_client/hyperdome_client.py [hunk 1]
```diff
@@ -33,19 +33,25 @@
...
+import socket
+import requests
 
-session = requests.Session()
-# session.proxies = {'http': 'socks5://127.0.0.1:9050',
-#                    'https': 'socks5://127.0.0.1:9050'}
 
 class Server(object):
     def __init__(self, url='', uname='', passwd='', is_therapist=False):
+
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
```

File: hyperdome_client/hyperdome_client.py [hunk 2]
```diff
@@ -33,19 +33,25 @@
...
         self.url = url
+        self._check_url()
+        print(self.url)
         self.username = uname
         self.password = passwd
         self.is_therapist = is_therapist
+    
+    def _check_url(self):
+        if self.url.find('http://') is -1 and self.url.find('https://') is -1:
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
        if not self.url.startswith('http://') and not self.url.startswith('https://'):
```
Tiny bit faster and cleaner

File: hyperdome_client/hyperdome_client.py [hunk 3]
```diff
@@ -71,7 +77,7 @@ def __init__(self, common, onion, qtapp, app, filenames, config=False, local_onl
         self.qtapp = qtapp
         self.app = app
         self.local_only = local_only
-
+
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion

```
Remove trailing space

File: hyperdome_client/hyperdome_client.py [hunk 4]
```diff
@@ -204,9 +214,9 @@ def send_message(self):
                 self.chat_history.append("You: " + message)
                 self.on_history_added()
                 if self.server.is_therapist: # needs auth
-                    session.post(f"{self.server.url}/message_from_therapist",headers={"username":self.server.username, "password":self.server.password,"message":message})
+                    self.session.post(f"{self.server.url}/message_from_therapist",headers={"username":self.server.username, "password":self.server.password,"message":message})
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
                    self.session.post(f"{self.server.url}/message_from_therapist", headers={"username":self.server.username, "password":self.server.password, "message":message})
```
flake8 compliant. Doesn't matter too much.

File: hyperdome_client/hyperdome_client.py [hunk 5]
```diff
@@ -204,9 +214,9 @@ def send_message(self):
...
                 else: # normal user
-                    session.post(self.server.url + '/message_from_user', data = {'message':message, 'guest_id':self.uid} )
+                    self.session.post(self.server.url + '/message_from_user', data = {'message':message, 'guest_id':self.uid} )
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
                    self.session.post(f'{self.server.url}/message_from_user', data={'message':message, 'guest_id':self.uid} )
```
fstrings are generally faster and easier to read than string concatenation

File: hyperdome_client/hyperdome_client.py [hunk 6]
```diff
@@ -218,7 +228,20 @@ def on_history_added(self):
         self.chat_history = []
 
     def get_uid(self):
-        self.uid = session.get(self.server.url + '/generate_guest_id').text
+        try:
+            self.uid = self.session.get(self.server.url + '/generate_guest_id').text
+        except Exception as e:
+            raise e
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]: whAT
I 
don't understand the point of catching and then immediately re-raising

File: hyperdome_client/hyperdome_client.py [hunk 7]
```diff
@@ -227,31 +250,31 @@ def server_switcher(self, server):
...
+                self.session.post(f"{self.server.url}/therapist_signup", data={"masterkey":"megumin","username":self.server.username,"password":self.server.password})
             else:
                 self.get_uid()
-                self.therapist = session.post(f"{self.server.url}/request_therapist",  data={"guest_id":self.uid}).text
+                self.therapist = self.session.post(f"{self.server.url}/request_therapist",  data={"guest_id":self.uid}).text
                 if self.therapist:
                     self.is_connected = True
-        except:
+        except Exception as e:
+            print (e)
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
            print(''.join(traceback.format_exception(type(e), e, e.__traceback__)))
```
This is how you get the full trace

File: hyperdome_client/hyperdome_client.py [hunk 8]
```diff
@@ -33,19 +33,25 @@
 from .server_status import ServerStatus
 from .add_server_dialog import AddServerDialog
 
-import requests
 
+import socks
+import socket
+import requests
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py:38]:
```suggestion
import requests
import traceback
```
For displaying the trace in a comment below

File: hyperdome_client/hyperdome_client.py [hunk 9]
```diff
@@ -218,7 +228,20 @@ def on_history_added(self):
...
+            raise e
+
+    @property
+    def session(self):
+        if getattr(self, '_session', None) is None:
+            self._session = requests.Session()
+            if self.onion.is_authenticated():
+                socks_address, socks_port = self.onion.get_tor_socks_port()
+                self._session.proxies = {'http': f'socks5h://{socks_address}:{socks_port}',
+                                    'https': f'socks5h://{socks_address}:{socks_port}'}
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
                                         'https': f'socks5h://{socks_address}:{socks_port}'}                
```
Fix hanging indent

File: hyperdome_client/hyperdome_client.py [hunk 10]
```diff
@@ -227,31 +250,31 @@ def server_switcher(self, server):
...
 
 
     def add_server(self, url, nick, uname, passwd, is_therapist):
         self.server = Server(url, uname, passwd, is_therapist)
         self.servers[nick] = self.server
-        self.server.is_therapist = is_therapist
         try:
             if self.server.is_therapist:
                 pass
                 #TODO: authenticate the therapist here when that's a thing
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py:270]:
```suggestion
                 # TODO: authenticate the therapist here when that's a thing
```
flake8 thing

File: hyperdome_client/hyperdome_client.py [hunk 11]
```diff
@@ -227,31 +250,31 @@ def server_switcher(self, server):
...
             else:
-                session.get(url + '/generate_guest_id').text
+                self.session.get(self.server.url + '/generate_guest_id').text
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
                self.session.get(f'{self.server.url}/generate_guest_id').text
```

File: hyperdome_client/hyperdome_client.py [hunk 12]
```diff
@@ -227,31 +250,31 @@ def server_switcher(self, server):
...
             self.server_dropdown.addItem(nick)
             self.server_add_dialog.close()
         except Exception as e:
-            print(e.with_traceback())
-            Alert(self.common, f"server {url} is invalid", QtWidgets.QMessageBox.Warning, buttons=QtWidgets.QMessageBox.Ok)
+            print(e)
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
            print(''.join(traceback.format_exception(type(e), e, e.__traceback__)))
```

File: hyperdome_client/hyperdome_client.py [hunk 13]
```diff
@@ -1,8 +1,8 @@
 # -*- coding: utf-8 -*-
 """
-OnionShare | https://onionshare.org/
+Hyperdome
 
-Copyright (C) 2014-2018 Micah Lee <micah@micahflee.com>
+Copyright (C) 2019 Skyelar Craver <scravers@protonmail.com>
```
Comment (stevenpitts) [inline hyperdome_client/hyperdome_client.py]:
```suggestion
Copyright (C) 2019 Skyelar Craver <scravers@protonmail.com> and Steven Pitts <makusu2@gmail.com>
```

**Review by stevenpitts — CHANGES_REQUESTED** [review]
Looks good for the most part. 
I added some linting fixes, but ignored a lot, since Onionshare didn't appear to lint their code and a lot of the issues are just left over from that. 
The only thing that bothers me is catching the exception and then immediately re-raising it.
\o/

File: hyperdome_client/hyperdome_client.py (hunk 6 above)
Comment (arades79) [reply hyperdome_client/hyperdome_client.py]: because I deal with it in the calling funtion, as well as other exceptions that might happen?

Comment (stevenpitts) [reply hyperdome_client/hyperdome_client.py]: But how is that any different than just removing the try and except? Without it, the exception will still be raised and caught in the calling function

**Review by stevenpitts — APPROVED** [review]
Looks good! We can worry about minor formatting issues later, when we fix them for the rest of the code.

**Pull request merged**